pkg/domain/entities/types/containers.go

```diff
@@ -36,7 +37,42 @@ type ContainerStatsReport struct {
 }
 
 type ContainerUpdateOptions struct {
-	NameOrID                        string
+	NameOrID string
+	// Deprecated: Specgen should not be used to change the container configuration.
+	// Specgen is processed first, so values will be overwritten by values from other fields.
+	// To change configuration use other fields:
+	// Resources to change resource configuration, DevicesLimits to Limit device,
+	// RestartPolicy to change restart policy and RestartRetries to change restart retries.
 	Specgen                         *specgen.SpecGenerator
+	Resources                       *specs.LinuxResources
+	DevicesLimits                   *define.UpdateContainerDevicesLimits
 	ChangedHealthCheckConfiguration *define.UpdateHealthCheckConfig
+	RestartPolicy                   *string
```

Comment:
Any reason to define this `*string`? It seems pretty clear to me that `string` should work fine too. Just leave the string empty if it should not be changed sounds easy enough. I don't see what the extra pointer offers here? Allowing user to set the restart policy to an empty string seems pointless as this is not valid?

Reply:
I'm sure an empty string would work, but the intent was to be consistent in the representation of the unset value throughout the structure.